zerver/models.py

```diff
@@ -1401,11 +1401,24 @@ def get_status_dict_by_realm(realm_id):
         return UserPresence.get_status_dicts_for_query(query, mobile_user_ids)
 
     @staticmethod
+    def _update_aggregated_status(status_info, user_statuses):
+        # type: (Dict[str, Any], Optional[Dict[Any, Any]]) -> Dict[Any, Any]
+        # If one of the statuses is 'active', aggregation status is active
+        # and timestamp is taken from the oldest 'active' status. Otherwise,
+        # It gets 'idle' status and timestamp is taken from the latest 'idle' status.
+        if not user_statuses or user_statuses['aggregated']['status'] == 'idle':
+            return {
+                'status': status_info['status'],
+                'timestamp': status_info['timestamp']
+            }
+        return user_statuses['aggregated']
```

Comment:
I think you missed a (undocumented) detail of how this works -- a timestamp attached to "active" is the last time the user was _confirmed active_, not the time they started being active.  This is used for clients that briefly cache the user's status -- they can show "last active 1 minute ago" or "last active 15 minutes ago" correctly.  

So I think we always want to take the newest status?